File: backend/policies/policy_engine.py
BLOCKED_WORDS = [
    "hack bank",
    "steal password",
    "malware",
    "ransomware"
]

def evaluate_policy(task: str):

    for word in BLOCKED_WORDS:

        if word.lower() in task.lower():

            return {
                "allow": False,
                "risk_level": "high",
                "risk_score": 90,
                "action": "block",
                "reason": f"Matched blocked phrase: {word}"
            }

    return {
        "allow": True,
        "risk_level": "low",
        "risk_score": 0,
        "action": "allow",
        "reason": "allowed execution"
    }

File: backend/control_plane/policy_engine.ts
export const policyEngine = {
  async check(userId, prompt) {
    let riskScore = 0;

    if (prompt.includes("hack")) riskScore += 80;
    if (prompt.length > 2000) riskScore += 30;

    const allowed = riskScore < 70;

    return {
      allowed,
      riskScore,
      reason: allowed ? "OK" : "HIGH_RISK_CONTENT",
    };
  },
};